substra/cli/printers.py

```diff
@@ -218,8 +218,8 @@ def print(data, *args, **kwargs):
         print(yaml.dump(data, default_flow_style=False))
 
 
-class AlgoPrinter(AssetPrinter):
-    asset_name = 'algo'
+class BaseAlgoPrinter(AssetPrinter):
+    asset_name = None  # to be updated in child classes
```

Comment:
You could even remove this line, making it clear that the BaseAlgoPrinter isn't related to any asset.